src/Microsoft.ComponentDetection.Contracts/FileComponentDetectorWithCleanup.cs

```diff
@@ -151,4 +149,19 @@ private bool TryGetCleanupFileDirectory(ProcessRequest processRequest, out strin
 
         return false;
     }
+
+    private (bool Success, HashSet<string> Files, HashSet<string> Directories) TryGetFilesAndDirectories(string root, IList<string> patterns, int depth)
+    {
+        try
+        {
+            var (files, directories) = this.DirectoryUtilityService.GetFilesAndDirectories(root, patterns, depth);
+            return (true, files, directories);
+        }
+        catch (UnauthorizedAccessException e)
+        {
+            // ignore files and directories that we don't have access to
```

Comment:
can we mock this exception and make sure that this is covered in a unit test ?

Reply:
done